test_scripts/API/GetWayPoints/commonGetWayPoints.lua

```diff
@@ -0,0 +1,284 @@
+
+--[[ Local Variables ]]
+local ptu_table = {}
+local hmiAppIds = {}
+
+local commonGetWayPoints = {}
+
+commonGetWayPoints.timeout = 2000
+commonGetWayPoints.minTimeout = 500
+commonGetWayPoints.DEFAULT = "Default"
+
+local function checkIfPTSIsSentAsBinary(bin_data)
+  if not (bin_data ~= nil and string.len(bin_data) > 0) then
+    commonFunctions:userPrint(31, "PTS was not sent to Mobile in payload of OnSystemRequest")
+  end
+end
+
+function commonGetWayPoints.getGetWayPointsConfig()
+  return {
+    keep_context = false,
+    steal_focus = false,
+    priority = "NONE",
+    default_hmi = "NONE",
+    groups = { "Base-4", "GetWayPoints" }
+  }
+end
+
+local function getPTUFromPTS(tbl)
+  tbl.policy_table.consumer_friendly_messages.messages = nil
+  tbl.policy_table.device_data = nil
+  tbl.policy_table.module_meta = nil
+  tbl.policy_table.usage_and_error_counts = nil
+  tbl.policy_table.functional_groupings["DataConsent-2"].rpcs = json.null
+  tbl.policy_table.module_config.preloaded_pt = nil
+  tbl.policy_table.module_config.preloaded_date = nil
+end
+
+local function jsonFileToTable(file_name)
+  local f = io.open(file_name, "r")
+  local content = f:read("*all")
+  f:close()
+  return json.decode(content)
+end
+
+local function ptu(self, id, ptu_update_func)
+  local function getAppsCount()
+    local count = 0
+    for _ in pairs(hmiAppIds) do
+      count = count + 1
+    end
+    return count
+  end
+
+  local policy_file_name = "PolicyTableUpdate"
+  local policy_file_path = commonFunctions:read_parameter_from_smart_device_link_ini("SystemFilesPath")
+  local pts_file_name = commonFunctions:read_parameter_from_smart_device_link_ini("PathToSnapshot")
+  local ptu_file_name = os.tmpname()
+  local requestId = self.hmiConnection:SendRequest("SDL.GetURLS", { service = 7 })
+  EXPECT_HMIRESPONSE(requestId)
+  :Do(function()
+      self.hmiConnection:SendNotification("BasicCommunication.OnSystemRequest", { requestType = "PROPRIETARY", fileName = pts_file_name })
+      getPTUFromPTS(ptu_table)
+       local function updatePTU(tbl)
+        tbl.policy_table.functional_groupings["GetWayPoints"] = {
+          rpcs = {
+            GetWayPoints = {
+               hmi_levels = { "BACKGROUND", "FULL", "LIMITED", "NONE" }
+            }
+          }
+        }
+        tbl.policy_table.app_policies[commonGetWayPoints.getMobileAppId(id)] = commonGetWayPoints.getGetWayPointsConfig()
+      end
+      updatePTU(ptu_table)
+      if ptu_update_func then
+        ptu_update_func(ptu_table)
+      end
+      local function tableToJsonFile(tbl, file_name)
+        local f = io.open(file_name, "w")
+        f:write(json.encode(tbl))
+        f:close()
+      end
+      tableToJsonFile(ptu_table, ptu_file_name)
+
+      local event = events.Event()
+      event.matches = function(self, e) return self == e end
+      EXPECT_EVENT(event, "PTU event")
+      :Timeout(11000)
+
+      for id = 1, getAppsCount() do
```

Comment:
@LitvinenkoIra There is a shadowing of function argument "id". It's better to have different names.